### PR TITLE
Jackyangzzh/shatter bottle

### PR DESCRIPTION
**Pull request by jackyangzzh**
Shatter bottle feature
- Bottle will shatter when drop from a distance
- Can hide object inside the bottle so that it doesn't shatter along with the bottle

### CODE REVIEW

**Comment by dilmerv** [comment]
This looks awesome @jackyangzzh !

Looks like there may be an issue with the source model, I keep getting this error after I throw the object:

Failed to create Convex Mesh from source mesh "Potion1_Cube.001_cell.012_cell.005". An internal unspecified error has occured that could mean the Quickhull algorithm found the input mesh topologically challenging. 
UnityEngine.Object:Instantiate<UnityEngine.GameObject> (UnityEngine.GameObject,UnityEngine.Vector3,UnityEngine.Quaternion)
ShatterFeature:Fracture () (at Assets/Scenes/EscapeRoom/Scripts/Behaviours/ShatterFeature.cs:32)
ShatterFeature:OnCollisionEnter (UnityEngine.Collision) (at Assets/Scenes/EscapeRoom/Scripts/Behaviours/ShatterFeature.cs:19)
UnityEngine.Physics:OnSceneContact (UnityEngine.PhysicsScene,intptr,int)

Also the prefabs may have broken references to materials:
image

**Comment by jackyangzzh** [comment]
Ah yes the bug is expected. Potions 2 and 3 would throw errors because I haven't created fracture versions of them. I removed those two models from the project to avoid confusion. Let me know if you'd like me to add fracture support for them!

**Comment by dilmerv** [comment]
Thanks @jackyangzzh I will review this and let you know !

File: Assets/Prefabs/XR Origin (Player).prefab [hunk 1]
```diff
@@ -1188,7 +1188,7 @@ LineRenderer:
   - {x: 0, y: 0, z: 1}
   m_Parameters:
     serializedVersion: 3
-    widthMultiplier: 0.005
```
Comment (dilmerv) [inline Assets/Prefabs/XR Origin (Player).prefab]: @jackyangzzh are these changes needed for this prefab?

File: Assets/Prefabs/XR Origin (Player).prefab [hunk 2]
```diff
@@ -1199,8 +1199,17 @@ LineRenderer:
         outSlope: 0
         tangentMode: 0
         weightedMode: 0
-        inWeight: 0.33333334
-        outWeight: 0.33333334
+        inWeight: 0
```
Comment (dilmerv) [inline Assets/Prefabs/XR Origin (Player).prefab]: Same with these ones below?

Comment (jackyangzzh) [reply Assets/Prefabs/XR Origin (Player).prefab]: Good catch. These changes shouldn't be there! Reverted in the new commit
